packages/next/client/index.js

```diff
@@ -14,13 +14,10 @@ import { isDynamicRoute } from '../next-server/lib/router/utils/is-dynamic'
 
 /// <reference types="react-dom/experimental" />
 
-// Polyfill Promise globally
-// This is needed because Webpack's dynamic loading(common chunks) code
-// depends on Promise.
-// So, we need to polyfill it.
-// See: https://webpack.js.org/guides/code-splitting/#dynamic-imports
-if (!window.Promise) {
-  window.Promise = require('@babel/runtime-corejs2/core-js/promise')
+// Polyfill finally only as Promise is polyfilled in the polyfills chunk
+// Finally is not supported in all module browsers
+if (!window.Promise.prototype.finally) {
```

Comment:
Save some bytes with self?
```suggestion
if (!self.Promise.prototype.finally) {
```